cmd/docker/docker.go

```diff
@@ -325,6 +326,25 @@ func tryPluginRun(ctx context.Context, dockerCli command.Cli, cmd *cobra.Command
 	return nil
 }
 
+// registerForceExitGoroutine registers a goroutine that will force exit the
+// process after 3 SIGTERM/SIGINT signals.
+func registerForceExitGoroutine(ctx context.Context, w io.Writer) {
+	// setup a signal handler to force exit after 3 SIGTERM/SIGINT
+	go func() {
+		<-ctx.Done()
+		sig := make(chan os.Signal, 2)
+		signal.Notify(sig, platformsignals.TerminationSignals...)
+		count := 0
+		for range sig {
+			count++
+			if count >= 2 {
+				_, _ = fmt.Fprint(w, "\ngot 3 SIGTERM/SIGINTs, forcefully exiting\n")
+				os.Exit(1)
+			}
+		}
```

Comment:
This can be simplified into:

```suggestion
		for i := 0; i < 3; i++ {
		   <-sig
		}
		_, _ = fmt.Fprintln(w, "\ngot 3 SIGTERM/SIGINTs, forcefully exiting")
		os.Exit(1)
```